### ISSUE TITLE
azure - get service properties for tables/queues/blob

### ISSUE THREAD

**Issue by thisisshi** [cloud-custodian/cloud-custodian#2889]
azure.storage resource doesn't return logging or metrics information on queues/tables/blob, need new filter to get service properties back

proposed syntax:
```yaml
policies:
  - name: logging-on-az-queues
    resource: azure.storage
    filters:
      - type: queue-service-properties
        key: logging.read
        value: true
      - type: queue-service-properties
        key: logging.write
        value: true
      - type: queue-service-properties
        key: logging.delete
        value: true

  - name: logging-on-az-blob
    resource: azure.storage
    filters:
      - type: blob-service-properties
        key: logging.read
        value: true
      - type: blob-service-properties
        key: logging.write
        value: true
      - type: blob-service-properties
        key: logging.delete
        value: true

```

**Comment by thisisshi** [comment]
for storage-account tables:
```python
from azure.cosmosdb.table.tableservice import TableService
table_service = TableService(account_name='some-storage'', account_key='some-key')
props = table_service.get_table_service_properties()
```

**Comment by stefangordon** [comment]
Yeah so this is the Storage Analytics Diagnostics settings, or what you'd see under `Diagnostics (Classic)` in the portal.  It is weird, at first glance it appears you can get the Storage Analytics Diagnostics settings for Blob Service from the management plane (`BlobServicesOperations .get_service_properties`) but the queue/table only from the data plane.    

We have the code in storage utilities to create RBAC based credentials to storage accounts to create data clients so we could do this in a value filter subclass pretty easy I think.   For each data client we just call `get_service_properties` or similar.  

We could also just make a dedicated Storage diagnostics filter with purpose built attributes that accept bools to query for different settings, if its useful.  Whoever implements this - perhaps make a recommendation on the issue and get feedback before you go too far finalizing filters.

**Comment by erwelch** [comment]
This was addressed in #4222 - will close.
